packages/calling-component-bindings/src/baseSelectors.ts

```diff
@@ -45,11 +46,9 @@ export const getDominantSpeakers = (
 export const getRemoteParticipants = (
   state: CallClientState,
   props: CallingBaseSelectorProps
-):
-  | undefined
-  | {
-      [keys: string]: RemoteParticipantState;
-    } => state.calls[props.callId]?.remoteParticipants;
+): undefined | RemoteParticipantState[] => {
+  return updateUserDisplayNames(state.calls[props.callId]?.remoteParticipants);
```

Comment:
cc. @prprabhu-ms - you mentioned not wanting to augment the base selectors -- should this be moved to a secondary selector `getRemoteParticipantsWithUpdatedDisplayNames`?

Reply:
(reason for it to be a base selector/its own selector - we want everyone getting a displayName to get a consistent display name across all components)

Reply:
I mean it would be a bigger diff but we could just have this pipe into a new base selector if the concern interacting with whats coming out of the call state at this point?

Reply:
Base selectors should stay simple. They _only_ figure out what part of `state` to return and then return it. No modifications. Reasons.

* Clarity in what a base selector is.
* A base selector runs on *every* React render pass. This means that any functions you call here are called on every render pass (even if they memoize their returned values, they will be called each time)
* The most insidious: You have to be very careful to memoize the returned values yourself. Other selectors don't run if their inputs don't change. So if you fail to memoize return values, it means that they have a snowball effect on what gets updated upstream, *but only when their inputs change*. For base selectors that is on every single render pass. So the cost of failure to memoize (James pointed out an example) is much higher.


As an aside, business logic "use phone number as display name" is likely better kept closer to the UI layer (i.e., each components' top-level-ish selector should make this determination.